bin/ssh_scan

```diff
@@ -48,14 +50,30 @@ opt_parser = OptionParser.new do |opts|
     options[:timeout] = timeout.to_i
   end
 
+  opts.on("-O", "--from_json [FilePath]",
+          "File to read JSON output from") do |file|
+    unless File.exists?(file)
+      puts "\nReason: Invalid file"
+      exit
+    end
+    file = open(file)
+    json = file.read
+    parsed_json = JSON.parse(json)
+    parsed_json.each do |host|
+      options[:sockets] += target_parser.enumerateIPRange(host['ip'], host['port'])
+    end
+  end
+
   opts.on("-o", "--output [FilePath]",
           "File to write JSON output to") do |file|
     $stdout.reopen(file, "w")
   end
 
   opts.on("-p", "--port [PORT]",
           "Port (Default: 22)") do |port|
-    options[:port] = port.to_i
+    socket = options[:sockets].shift
+    ip = socket.chomp.split(':').shift
```

Comment:
@royalharsh @claudijd I have a doubt here with -p as an argument we directly pass port we don't have ip.